MetaU Capstone Project/API/SpotifyManager.m

```diff
@@ -114,15 +115,27 @@ - (void)playerStateDidChange:(nonnull id<SPTAppRemotePlayerState>)playerState {
     NSLog(@"Track name: %@", playerState.track.name);
     NSLog(@"player state changed");
 
+    @try {
+        self.currentTrack = [self currentTrackInfo:playerState.track];
+        self.currentTrackContentItem = playerState.track;
+    } @catch (NSException *exception) {
+        NSLog(@"Error getting current track...");
+    }
+    
+    [self recentlyPlayedTrack];
+
+    NSLog(@"PREVIOUS TRACK: %@", self.previousTrack);
+    NSLog(@"CURRENT TRACK : %@", self.currentTrack);
+
     [self sendNotification];
 }
 
 - (void)sendNotification {
     [[NSNotificationCenter defaultCenter] postNotificationName:@"playerStateChangeNotification" object:self];
 }
 
-- (id<SPTAppRemoteTrack>) getCurrentTrackInfo {
-    return self.currentTrack;
+- (void)sendNewTrackNotification {
+    [[NSNotificationCenter defaultCenter] postNotificationName:@"newTrackNotification" object:self];
```

Comment:
You should make these notification names constants. I've seen a few so far:
- `@"newTrackNotification"`
- `@"playerStateChangeNotification"`

The reason is that its possible to misspell these. And defining it as a constant provides code completion instead of typing it out.

So for example it would be like this:

```
static NSString *const newTrackNotificationName = @"newTrackNotification";

[[NSNotificationCenter defaultCenter] postNotificationName:newTrackNotificationName object:self];

```